lumen/ai/ui.py

```diff
@@ -567,6 +567,7 @@ def sync_available_sources_memory(_, __, sources):
 
             async def render_output(_, old, new):
                 added = [out for out in new if out not in old]
+                index = self._explorations.active if len(self._explorations) else -1
```

Comment:
Actually I do not believe this to be correct. Seems like it'll break if you switch tabs while an exploration is running.

Reply:
Fixed.